File: payment_service/src/services/jwt_service.js
import jwt from 'jsonwebtoken';

const JWT_SECRET = process.env.JWT_SECRET;
if (!JWT_SECRET) console.error('JWT_SECRET is not set in .env file');

export default class JwtService {

    /**
     * @function verify
     * @param {String} token
     * @returns {Promise<Object>}
     */
    static async verify(token) {
        if (!token) throw new Error('Token is required');
        if (typeof token !== 'string') throw new Error('Token must be a string');
        
        return jwt.verify(token, JWT_SECRET);
    }
}

File: embedding_service/src/services/jwt_service.py
from fastapi import Request, HTTPException
import jwt
import os
from dotenv import load_dotenv

load_dotenv()

SECRET_KEY = os.getenv("JWT_SECRET")
if SECRET_KEY is None: print("JWT_SECRET is not set in .env")

def verify(token):
    if SECRET_KEY is None:
        raise Exception("JWT_SECRET is not set in .env")
    if token is None: raise Exception("Token is required")
    return jwt.decode(token, SECRET_KEY, algorithms=["HS256"]) 

def generate(_id, role):
    if SECRET_KEY is None:
        raise Exception("JWT_SECRET is not set in .env")
    if _id is None: raise Exception("_id is required")
    if role is None: raise Exception("Role is required")
    payload = {"_id": _id, "role": role}
    token = jwt.encode(payload, SECRET_KEY, algorithm="HS256")
    return token